### ISSUE TITLE
Can not Install: Missing query-engine-debian-openssl-1.1.x

### ISSUE THREAD

**Issue by KyleTryon** [prisma/prisma#1642]
**Env info:**

Running Ubuntu 18.04 in Windows 10 WSL 2.

```
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.1 LTS
Release:        18.04
Codename:       bionic
```

Globally installed prisma2:
`prisma2@2.0.0-preview022, binary version: f608db3b9e4f50f9a384356e532357b0a525969a`

**package.json**

```
  "devDependencies": {
    "prisma2": "^2.0.0-preview022",
    "ts-node": "^8.6.2",
    "typescript": "^3.7.5"
  },
  "dependencies": {
    "@prisma/client": "^2.0.0-preview022"
  }
```

**Error**

```
> @prisma/client@2.0.0-preview022 postinstall [project]/node_modules/@prisma/client
> node scripts/postinstall.js

Error: Error: Command failed with ENOENT:[project]/node_modules/prisma2/query-engine-debian-openssl-1.1.x --enable_raw_queries cli 
--dmmf
spawn [project]/node_modules/prisma2/query-engine-debian-openssl-1.1.x ENOENT
```

**Reproduce:**

Run: npm install

```
> prisma2@2.0.0-preview022 install [project]/node_modules/prisma2
> node download-build/index.js

> Downloading Prisma engines for debian-openssl-1.1.x [====================] 100%


> @prisma/client@2.0.0-preview022 postinstall [project]/node_modules/@prisma/client
> node scripts/postinstall.js

Error: Error: Command failed with ENOENT:[project]/node_modules/prisma2/query-engine-debian-openssl-1.1.x --enable_raw_queries cli 
--dmmf
spawn [project]/node_modules/prisma2/query-engine-debian-openssl-1.1.x ENOENT
npm WARN postsnail-core@1.0.0 No repository field.

added 11 packages from 42 contributors and audited 11 packages in 7.457s
found 0 vulnerabilities
```

You can see `debian-openssl-1.1.x ` was apparently saved successfully but after the operation, I do not see it in the `node_modules/prisma2` directory:

```
[user]@DESKTOP: [project]/node_modules/prisma2$ ls -lah
total 12K
drwxrwxrwx 1 [user] [user]     512 Feb 18 17:22 .
drwxrwxrwx 1 [user] [user]     512 Feb 18 17:22 ..
-rwxrwxrwx 1 [user] [user]     1.1K Oct 26  1985 README.md
drwxrwxrwx 1 [user] [user]     512 Feb 18 17:22 build
drwxrwxrwx 1 [user] [user]     512 Feb 18 17:22 download-build
-rwxrwxrwx 1 [user] [user]      4 Feb 18 17:22 download-done
-rwxrwxrwx 1 [user] [user]      4.1K Feb 18 17:22 package.json
drwxrwxrwx 1 [user] [user]     512 Feb 18 17:22 prisma-client
```

Running as sudo will produce the same error.

**Comment by janpio** [comment]
Quickly tried to reproduce this on WSL (1), and could not - there the binaries were successfully downloaded and a Prisma Client could be generated with the steps mentioned in https://github.com/prisma/prisma2/blob/master/docs/getting-started/start-from-scratch-with-empty-db/quickstart-prisma-migrate.md (which you linked to in Slack).

@pantharshit00 please try to reproduce this with WSL2 (https://docs.microsoft.com/en-us/windows/wsl/wsl2-install).

(@KyleTryon If you are not depending on WSL, Prisma works fine in normal Windows. It will download different binaries, that should work just fine. Maybe that is a temporary workaround for you.)

**Comment by KyleTryon** [comment]
I uninstalled ubuntu and reinstalled it for a fresh WSL 2 box and tried again. This time I got a different error, then re-ran the installation with no error shown.

```
user@DESKTOP:[project]$ sudo npm install

> prisma2@2.0.0-preview022 install [project]/node_modules/prisma2
> node download-build/index.js

> Downloading Prisma engines for debian-openssl-1.1.x [====================] 100%




> @prisma/client@2.0.0-preview022 postinstall [project]/node_modules/@prisma/client
> node scripts/postinstall.js

GeneratorError: Error: EPERM: operation not permitted, copyfile '[project]/node_modules/prisma2/query-engine-debian-openssl-1.1.x' -> '[project]/node_modules/@prisma/client/runtime/query-engine-debian-openssl-1.1.x'
    at GeneratorProcess.handleResponse ([project]/node_modules/prisma2/build/index.js:2:177310)
    at LineStream.<anonymous> ([project]/node_modules/prisma2/build/index.js:2:176926)
    at LineStream.emit (events.js:321:20)
    at LineStream.EventEmitter.emit (domain.js:485:12)
    at addChunk (_stream_readable.js:297:12)
    at readableAddChunk (_stream_readable.js:273:9)
    at LineStream.Readable.push (_stream_readable.js:214:10)
    at LineStream.Transform.push (_stream_transform.js:152:32)
    at LineStream._pushBuffer ([project]/node_modules/prisma2/build/index.js:2:840724)
    at LineStream._transform ([project]/node_modules/prisma2/build/index.js:2:840543) {
  code: -32000,
  data: null
}

You can now start using Prisma Client in your code:

'''
import { PrismaClient } from '@prisma/client'
// or const { PrismaClient } = require('@prisma/client')

const prisma = new PrismaClient()
'''

Explore the full API: http://pris.ly/d/client

^C
user@DESKTOP:[project]$ sudo npm install
npm WARN postsnail-core@1.0.0 No repository field.

audited 11 packages in 0.47s
found 0 vulnerabilities

user@DESKTOP:[project]$ 

```

Everything is now working fine 👍  Was able to run:

```
npx prisma2 migrate save --name 'init' --experimental
npx prisma2 migrate up --experimental
npx prisma2 generate
npx ts-node index.ts
```

Thanks everyone 🙏

**Comment by janpio** [comment]
Hm, we might want to investigate this anyway - the `EPERM` error also is not the best experience ;)

**Comment by TSTsankov** [comment]
sudo npm i -g prisma2

**Comment by TheAifam5** [comment]
Same problem on ArchLinux.

**Comment by unlight** [comment]
Same problem inside docker
```
Error: Error: Command failed with ENOENT: /app/node_modules/prisma2/query-engine-debian-openssl-1.1.x --enable-raw-queries cli dmmf
spawn /app/node_modules/prisma2/query-engine-debian-openssl-1.1.x ENOENT
```

**Comment by janpio** [comment]
Also on Windows an WSL @unlight? If not, please open a new issue and include all available information so we can try to reproduce this.

**Comment by pantharshit00** [comment]
Can you all please try again with the latest version `preview024`?

**Comment by krn0x2** [comment]
@pantharshit00 The `preview024` is also broken with docker. It gives the same error.

**Comment by pantharshit00** [comment]
@krn0x2 Can you please your dockerfile here?

**Comment by andriisoldatenko** [comment]
same issue for alpine 3.11 and `2.0.0-preview024` and `2.0.0-preview025`:

`$ cat Dockerfile:`
```
RUN apk add --no-cache --virtual .build-deps \
		build-base \
		git \
		python \
	&& apk add --no-cache \
		nodejs-current \
		nodejs-npm \
		openssl \
	&& npm install \
	&& apk del .build-deps
```

**docker build and docker run:**
```
Error: Error: Command failed with ENOENT: /houston/node_modules/prisma2/query-engine-debian-openssl-1.1.x --enable-raw-queries cli dmmf
spawn /houston/node_modules/prisma2/query-engine-debian-openssl-1.1.x ENOENT
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! houston-api@1.0.0 build: `npx prisma2 generate && babel src -d dist --copy-files`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the houston-api@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-03-27T15_10_42_738Z-debug.log
```

```
node -v
v13.8.0
npm --version
6.13.7
```

```
cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.11.3
PRETTY_NAME="Alpine Linux v3.11"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"

openssl version
OpenSSL 1.1.1d  10 Sep 2019
```

**Comment by andriisoldatenko** [comment]
@pantharshit00  please take a look ^

**Comment by andriisoldatenko** [comment]
if i add to my dockerfile:

```
# Download and install glibc
RUN apk add --update curl && \
	curl -Lo /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub && \
	curl -Lo glibc.apk "https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VERSION}/glibc-${GLIBC_VERSION}.apk" && \
	curl -Lo glibc-bin.apk "https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VERSION}/glibc-bin-${GLIBC_VERSION}.apk" && \
	apk add glibc-bin.apk glibc.apk && \
	/usr/glibc-compat/sbin/ldconfig /lib /usr/glibc-compat/lib && \
	echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf && \
	apk del curl && \
	rm -rf glibc.apk glibc-bin.apk /var/cache/apk/*
```
from https://github.com/prisma/prisma2/issues/702#issuecomment-569778198

**it works!**

**Comment by pantharshit00** [comment]
@andriisoldatenko We do not compile for alpine yet so I would advice you to use a different docker image.

**Comment by yleigh** [comment]
I tried andriisoldatenko suggested above with node:12-alpine, however doesn't work for me.

**Comment by yleigh** [comment]
@pantharshit00 I tried node:slim, it works for my nodejs app with prisma2. However, the image is still kind of big. Is there a way that node:apline could work for nodesjs app with prisma2 Docker build? What missing dependices should I add?
For node:12-alpine, my Docker build got error:
`Step 27/31 : RUN npx prisma2 generate
 ---> Running in fb0918ffb9ed
Error: Error: Command failed with ENOENT: /usr/local/lib/node_modules/@prisma/cli/query-engine-debian-openssl-1.1.x --enable-raw-queries cli dmmf
spawn /usr/local/lib/node_modules/@prisma/cli/query-engine-debian-openssl-1.1.x ENOENT
The command '/bin/sh -c npx prisma2 generate' returned a non-zero code: 1`

I even added
`RUN apk update && apk add openssl`
at step 22.

**Comment by pantharshit00** [comment]
@yleigh You will need to add `glibc` as binaries are based on that. Alpine only comes with `musl` and rust with musl is not that straight forward so it may take some time for us to release alpine support so no timeline on that.

A community member posted steps that worked for them here: https://github.com/prisma/prisma/issues/702#issuecomment-569778198

or you can compile the binaries for Alpine as one of the other community members did: https://github.com/prisma/prisma/issues/702#issuecomment-599121308

**Comment by Tenkir** [comment]
Also having the same issue inside `node:13` and `node:13-slim`

**Comment by pantharshit00** [comment]
@Tenkir Can you please share your Dockerfile?

**Comment by emroot** [comment]
Getting the same issue, it's trying to download `musl-openssl-1.1.x` from this url: `https://binaries.prisma.sh/master/afd294205618b1c825b013ba6f5a6ebe4aa4a514/musl-openssl-1.1.x/query-engine.gz` but the url returns `NoSuchKey` error
![image](https://user-images.githubusercontent.com/110190/81769683-19ab9500-9493-11ea-9e2e-0b476cbebf26.png)

**Comment by yleigh** [comment]
With node:slime, in Dockerfile I added this and fixed the issue:
`RUN apt-get -qy update && apt-get -qy install openssl`

**Comment by pantharshit00** [comment]
I am tested this out and it works:
![image](https://user-images.githubusercontent.com/22195362/81820269-03164400-954e-11ea-8171-7d859e7ecf18.png)


I used the following Dockerfile:
```
FROM node:12-alpine

ENV DEBUG *


RUN npm install -g @prisma/cli

RUN prisma version
```

Also, the binaries are not stored at `https://binaries.prisma.sh/master/afd294205618b1c825b013ba6f5a6ebe4aa4a514/musl-openssl-1.1.x/query-engine.gz` that scheme is wrong. Url from the correct schema: `https://binaries.prisma.sh/master/0c2ec197653b278b2978845ef958db88824cd82e/linux-musl/migration-engine.gz`

**Comment by husseinraoouf** [comment]
I have the same issue in nixos

when trying to run `npx prisma migrate save --name init --experimental`
```
Error: Get config Error: Command failed with ENOENT: [project folder]/node_modules/@prisma/cli/query-engine-debian-openssl-1.1.x cli get-config
spawn [project folder]/node_modules/@prisma/cli/query-engine-debian-openssl-1.1.x ENOENT
```

**Comment by pantharshit00** [comment]
@husseinraoouf, Please open a new issue with possibly a reproduction and details about your OS

**Comment by DosImpact** [comment]
same issue in docker
dockerfile

Error Message
`
root@8bcd4d7dfab7:/usr/src/app# sudo npx prisma generate
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Error: Get config /usr/src/app/node_modules/@prisma/cli/query-engine-debian-openssl-1.1.x: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory
`

`
FROM node:10.20.1-slim@sha256:05d1d270480b6e99753076b6656bb5a37edb7ca31af20c008568a556bc82d2a8

RUN  apt-get update 

RUN mkdir -p /usr/src/app
WORKDIR /usr/src/app
ENV PATH=/usr/src/app/node_modules/.bin:$PATH


COPY package.json /usr/src/app/package.json
RUN npm install --silent
COPY prisma /usr/src/app/prisma
COPY . /usr/src/app



EXPOSE 4001

CMD npm run start:dev


`

**Comment by Jolg42** [comment]
Closing as old.

Please open a new issue if you experience this with a more recent version of Prisma.

**Comment by gustawdaniel** [comment]
In my case `arch` these commands helped:

```
rm -rf node_modules
npm i
npx prisma generate
```

**Comment by saharshg** [comment]
Downloading Prisma engines for Node-API for debian-openssl-1.1.x [                    ] 0%Error: request to https://binaries.prisma.sh/all_commits/ceb5c99003b99c9ee2c1d2e618e359c14aef2ea5/debian-openssl-1.1.x/prisma-fmt.gz.sha256 failed, reason: unable to verify the first certificate


Still not resolved in prisma version 4.9.0, Ubuntu 20.04.5 LTS

**Comment by Jolg42** [comment]
@saharshg I think I might be able to help you, but could you open [a new issue here first](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml)?
Please make sure to mention your Node.js version and environment info in the template.
